python/ray/tests/test_cancel.py

```diff
@@ -306,5 +311,32 @@ def many_resources():
     assert ray.get(many_fut, timeout=30)
 
 
+def test_cancel_list(ray_start_regular):
+    @ray.remote
+    def blocking_operation_1(x):
+        time.sleep(x)
+
+    @ray.remote
+    def blocking_operation_2(x):
+        time.sleep(x)
+
+    obj_ref1 = blocking_operation_1.remote(100)
+    obj_ref2 = blocking_operation_2.remote(200)
+    obj_ref3 = blocking_operation_1.remote(100)
+    obj_refs = [obj_ref1, obj_ref2, obj_ref3]
+
+    # let the tasks start properly
+    time.sleep(30)
+
+    ray.cancel(obj_refs)
+
+    for obj_ref in obj_refs:
+        try:
+            ray.get(obj_ref)
+            assert False, f"{obj_ref} isn't cancelled."
+        except (RayTaskError, TaskCancelledError):
+            assert True
+
+
```

Comment:
can we add a unit test for partial failure?

Reply:
I am not sure if I understand this correctly. I think you mean to check on tasks which are already completed as inputs to `ray.cancel`. Am I right?

Reply:
Right, we need to test to cancel a list of tasks where some of them are already finished